applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_test_factory.py

```diff
@@ -0,0 +1,260 @@
+            self.adjoint_analysis.Initialize()
+
+    def verify_linear_solver(self, parameter_file):
+        # The mechanical solver selects automatically the fastest linear-solver available
+        # this might not be appropriate for a test, therefore in case nothing is specified,
+        # the previous default linear-solver is set
+        if not parameter_file["solver_settings"].Has("linear_solver_settings"):
+            # check if running in MPI because there we use a different default linear solver
+            if IsDistributedRun():
+                default_lin_solver_settings = KratosMultiphysics.Parameters("""{
+                    "solver_type" : "amesos",
+                    "amesos_solver_type" : "Amesos_Klu"
+                }""")
+            else:
+                default_lin_solver_settings = KratosMultiphysics.Parameters("""{
+                    "solver_type": "EigenSolversApplication.sparse_lu"
+                }""")
+            parameter_file["solver_settings"].AddValue("linear_solver_settings", default_lin_solver_settings)
+
+        solver_type = parameter_file["solver_settings"]["linear_solver_settings"]["solver_type"].GetString()
+        solver_type_splitted = solver_type.split(".")
+        if len(solver_type_splitted) == 2:
+            # this means that we use a solver from an application
+            # hence we have to check if it exists, otherwise skip the test
+            app_name = solver_type_splitted[0]
+            solver_name = solver_type_splitted[1]
+            if not kratos_utilities.CheckIfApplicationsAvailable(app_name):
+                self.skipTest('Application "{}" is needed for the specified solver "{}" but is not available'.format(app_name, solver_name))
+
+    def solve_adjoint_problem(self):
+        with KratosUnittest.WorkFolderScope(".", __file__):
+            self.adjoint_analysis.RunSolutionLoop()
+
+    def tearDown(self):
+        # Within this location context:
+        with KratosUnittest.WorkFolderScope(".", __file__):
+            self.adjoint_analysis.Finalize()
+            kratos_utilities.DeleteFileIfExisting(self.problem_name + ".time")
+            kratos_utilities.DeleteFileIfExisting(self.model_part_name + ".h5")
+            kratos_utilities.DeleteFileIfExisting(self.model_part_name + "-1.0000.h5")
+            kratos_utilities.DeleteFileIfExisting(self.model_part_name + "-1.1000.h5")
+
+
+@KratosUnittest.skipUnless(has_hdf5_application,"Missing required application: HDF5Application")
+class TestAdjointSensitivityAnalysisBeamStructureLocalStress(AdjointSensitivityAnalysisTestFactory):
+    primal_file_name = "adjoint_sensitivity_analysis_tests/adjoint_beam_structure_3d2n/beam_test_parameters.json"
+    adjoint_file_name = "adjoint_sensitivity_analysis_tests/adjoint_beam_structure_3d2n/beam_test_local_stress_adjoint_parameters.json"
+
+    def test_execution(self):
+        self.solve_adjoint_problem()
```

Comment:
This seems a little strange, couldn't you do that in setup as well? What do you think?

Reply:
good hint. I modified it.